guides/plugins/plugins/content/cms/add-cms-element-via-admin-sdk.md

```diff
@@ -0,0 +1,352 @@
+
+This guide will also not explain how a custom component can be created in general, so head over to the official guide about creating a custom component to learn this first. It also won't go too much into detail about the Admin Extension SDK. For that, please have a look at the [Admin Extension SDK documentation](https://shopware.github.io/admin-extension-sdk/docs/guide/getting-started/installation).
+
+{% hint style="info" %}
+This example uses TypeScript, which is recommended, but not required for developing Shopware.
+{% endhint %}
+
+{% page-ref page="../../administration/app-base-guide.md" %}
+{% page-ref page="../../administration/add-custom-component.md" %}
+
+## Creating your custom element
+
+Like in the Plugin Guide for this topic, imagine you want to create a new element to display a Dailymotion video. The shop manager can configure the link of the video to be shown. That's exactly what you're going to build in this guide.
+
+Creating a new element via app requires you to use the Admin Extension SDK.
+
+### Target structure
+
+It is yours to decide which approach to use when creating apps, since everything we're doing here will be loaded via iFrame, but since Shopware's best practice is a full Vue.js approach, this tutorial will do exactly that.
+
+When our extension is finished, you'll get the following file structure:
+
+{% code title="<plugin root>/src/Resources/app/administration/src" %}
+
+```bash
+├── base
+│   └── mainCommands.ts
+├── main.ts
+├── viewRenderer.ts
+└── views
+    └── swag-dailymotion
+        ├── swag-dailymotion-config.ts
+        ├── swag-dailymotion-element.ts
+        └── swag-dailymotion-preview.ts
+```
+
+{% endcode %}
+
+## Initial loading of components
+
+Everything starts in the `main.ts` file:
+
+```ts
+import 'regenerator-runtime/runtime';
+import { location } from '@shopware-ag/admin-extension-sdk';
+
+// Only execute extensionSDK commands when
+// it is inside a iFrame (only needed for plugins)
+if (location.isIframe()) {
+    if (location.is(location.MAIN_HIDDEN)) {
+        // Execute the base commands
+        import('./base/mainCommands');
+    } else {
+        // Render different views
+        import('./viewRenderer');
+    }
+}
+```
+
+This is main file, which is executed first and functions as our entry point.
+
+We start with `if(location.isIframe())` to make sure we only load content used inside iFrames. The reason for this is that the SDK can be used in apps and plugins, so this check makes sure that your code is executed in the right place.
+
+Next we need `if(location.is(location.MAIN_HIDDEN))` to **load the main commands**, which are defined in the `mainCommands.ts` file. This will only be used to load logic, but not templates into the Administration.
+
+Lastly the `else` case will be responsible for specific loading of views via `viewRenderer.ts`. This is where our view templates will be loaded.
```

Comment:
Replace the info a below:

This is the main file, which is executed first and functions as our entry point.

Start with `if(location.isIframe())` to make sure only content used inside iFrames is loaded. While the SDK is used in apps and plugins, this check ensures the code is executed in the right place.

Next you need `if(location.is(location.MAIN_HIDDEN))` to **load the main commands**, which are defined in the `mainCommands.ts` file. This will only be used to load logic, but not templates into the Administration.

Lastly, the `else` case will be responsible for specific loading of views via `viewRenderer.ts`. This is where the view templates will be loaded.